modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/DataGenerationApplication.java

```diff
@@ -20,27 +20,103 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import org.apache.ignite.IgniteCache;
 import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.internal.NodeStoppingException;
 import org.apache.ignite.internal.ducktest.utils.IgniteAwareApplication;
+import org.apache.ignite.internal.util.typedef.X;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
 
 /**
  *
  */
 public class DataGenerationApplication extends IgniteAwareApplication {
+    /** Logger. */
+    protected static final Logger log = LogManager.getLogger(DataGenerationApplication.class.getName());
+
+    /** */
+    private static final long DATAGEN_NOTIFY_INTERVAL_NANO = 1500 * 1000000L;
+
+    /** */
+    private static final int DATAGEN_NOTIFY_INTERVAL_AMOUNT = 10_000;
+
+    /** */
+    private static final int DELAYED_INITIALIZATION_AMOUNT = 10_000;
+
     /** {@inheritDoc} */
     @Override protected void run(JsonNode jsonNode) {
-        log.info("Creating cache...");
+        String cacheName = jsonNode.get("cacheName").asText();
+        boolean infinite = jsonNode.hasNonNull("infinite") && jsonNode.get("infinite").asBoolean();
+        boolean optimized = !jsonNode.hasNonNull("optimized") || jsonNode.get("optimized").asBoolean();
```

Comment:
how about "batched" or something like that?

Reply:
Why expose internal inplementation? It's not only buffered. It also choises nodes to load data by keys. It's an optimized load. What's the reason to name by one optimization step?

Reply:
the better case is to create 2 classes (DSDataGenerator and PutDG, or something like this) to make this simple. 
Just overwrite method from abstract DGApplication class)

Reply:
'optimized' got removed. It was involved to wrap node-stop-issue of the data streamer. The code got changed, I kept it as simple option which looks useful. Thi look to me trivilab, but if rised discussion, we don't need it in this ticket. For now, we need only contiguous loading. The streamer was introdused before. It is ok.